### PR TITLE
Implement custom hover for activity bar and panel

### LINKED ISSUE #118077: Use custom hover for activity bar tooltip
Kapture 2021-03-03 at 13 31 26

I think one of the areas that could benefit from quicker tooltips/custom hovers is the activity bar as it is a frequently used area and the native tooltip takes too long to load. We've seen this in user studies where new users are not sure what these items are and don't click on them.

cc @bpasero @alexr00

### PR DESCRIPTION
**Pull request by sandy081**
This PR fixes #118077

@Tyriar In order to support custom hovers for activity bar I have to do following changes to hover service:

- add `y` coordinate to target
- fix the service when `x` and `y` coordinates are passed. It seems I am the first one to use these coordinates.

Please review these changes

### CODE REVIEW

File: src/vs/workbench/services/hover/browser/hoverWidget.ts [hunk 1]
```diff
@@ -164,16 +164,22 @@ export class HoverWidget extends Widget {
 		const targetBounds = this._target.targetElements.map(e => e.getBoundingClientRect());
 
 		// Get horizontal alignment and position
-		let targetLeft = this._target.x !== undefined ? this._target.x : Math.min(...targetBounds.map(e => e.left));
-		if (targetLeft + this._hover.containerDomNode.clientWidth >= document.documentElement.clientWidth) {
```
Comment (Tyriar) [inline src/vs/workbench/services/hover/browser/hoverWidget.ts:168]: FYI this was meant to protect the consumer from displaying the hover in an invalid position.

**Review by Tyriar — APPROVED** [review]
Changes look fine, if we are going to do this elsewhere we should probably handle this via options instead, for example being able to specify left/right anchor position, vertical alignment and a gap.

File: src/vs/workbench/services/hover/browser/hoverWidget.ts (hunk 1 above)
Comment (sandy081) [reply src/vs/workbench/services/hover/browser/hoverWidget.ts:168]: Got it and it makes sense when `x` is being computed. If the value is coming from outside, I would assume its caller responsibility. In my case, this check is breaking the hover when sidebar is on the right. Hence removed this check when coordinates are passed.

**Comment by sandy081** [comment]
I agree it would be nice if there are options like you mentioned. In my case passing coordinates is good enough for now.

**Pull request merged**